src/components/donationFiat/Steps/FiatSubmit/Form/index.tsx

```diff
@@ -0,0 +1,150 @@
+import { PropsWithChildren, useState } from "react";
+import { useFormContext } from "react-hook-form";
+import { Link } from "react-router-dom";
+import { FiatDonateValues } from "../types";
+import { useRequestWidgetUrlMutation } from "services/apes";
+import CountrySelector from "components/CountrySelector";
+import Loader from "components/Loader";
+import { Label } from "components/form";
+import { useSetter } from "store/accessors";
+import { SubmitStep, setStep } from "slices/donation";
+import { humanize } from "helpers";
+import { appRoutes } from "constants/routes";
+import { getISOCountryCode } from "./countryList";
+import getBreakdown from "./getBreakdown";
+
+export default function Form(props: SubmitStep) {
+  const dispatch = useSetter();
+
+  function goBack() {
+    dispatch(setStep(props.step - 1));
+  }
+
+  const { token } = props.details;
+  const { id: endowId } = props.recipient;
+  const { fromBal, fromGift } = getBreakdown(token);
+  const {
+    handleSubmit,
+    resetField,
+    formState: { isValid },
+  } = useFormContext<FiatDonateValues>();
+  const [submitRequest] = useRequestWidgetUrlMutation();
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  async function submit(data: FiatDonateValues) {
+    setIsSubmitting(true);
+    const options = {
+      amount: +token.amount,
+      charityName: props.recipient.name,
+      countryCode: getISOCountryCode(data.country.name),
+      endowmentId: props.recipient.id,
+      sourceCurrencyCode: token.symbol,
+      splitLiq: props.details.pctLiquidSplit.toString(),
+    };
+    try {
+      const response: any = await submitRequest(options);
+      window.location.href = response.data.widgetUrl;
+    } catch (e) {
+      console.log(e);
+      setIsSubmitting(true);
+    }
+  }
+
+  return isSubmitting ? (
+    <div className="grid gap-4 rounded-md w-full">
+      <Loader
+        bgColorClass="bg-blue dark:bg-white"
+        gapClass="gap-2"
+        widthClass="w-4"
+      />
+    </div>
+  ) : (
+    <form
+      onSubmit={handleSubmit(submit)}
+      className="grid gap-4 rounded-md w-full"
+      autoComplete="off"
+    >
+      <h4 className="font-bold text-sm">Enter your payment details:</h4>
+      <div className="flex-col">
+        <Label className="mb-2" htmlFor="country">
+          Country of Origin *
```

Comment:
nit: make this more understandable/relatable for users.
```suggestion
          Country of Residence *
```

Reply:
Done